compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java

```diff
@@ -1461,61 +1479,71 @@ private boolean isEntryPointParamsInvalid(List<BLangVariable> params) {
         return false;
     }
 
-    private void visitInvokable(BLangInvokableNode invNode, SymbolEnv symbolEnv) {
+    /**
+     * Analyze the invokable body and identify the tainted status of the return value and the tainted status of
+     * parameters after the completion of the invokable. Based on that, attach the created taint table explaining
+     * possible taint outcomes of the function.
+     *
+     * @param invNode invokable node to be analyzed
+     * @param symbolEnv symbol environment for the invokable
+     * @return if the invocation is blocked due to an unanalyzed invocation
+     */
+    private boolean visitInvokable(BLangInvokableNode invNode, SymbolEnv symbolEnv) {
         if (analyzerPhase == AnalyzerPhase.LOOPS_RESOLVED_ANALYSIS || invNode.symbol.taintTable == null
                 || (invNode.getKind() == NodeKind.FUNCTION && ((BLangFunction) invNode).attachedOuterFunction)) {
             if (Symbols.isNative(invNode.symbol)
                     || (invNode.getKind() == NodeKind.FUNCTION && ((BLangFunction) invNode).interfaceFunction)) {
                 attachTaintTableBasedOnAnnotations(invNode);
-                return;
+                return false;
             }
             Map<Integer, TaintRecord> taintTable = new HashMap<>();
-            this.returnTaintedStatus = TaintedStatus.UNTAINTED;
+
             // Check the tainted status of return values when no parameter is tainted.
             analyzeAllParamsUntaintedReturnTaintedStatus(taintTable, invNode, symbolEnv);
             if (analyzerPhase == AnalyzerPhase.LOOP_ANALYSIS_COMPLETE) {
                 analyzerPhase = AnalyzerPhase.LOOP_ANALYSIS;
             }
-            boolean isBlocked = processBlockedNode(invNode, false);
+            boolean isBlocked;
+            if (mainFunctionAnalysis) {
+                isBlocked = processBlockedNode(invNode);
+            } else {
+                isBlocked = processBlockedNode(invNode);
+            }
```

Comment:
Both branches doing the same "isBlocked = processBlockedNode(invNode);" ?

Reply:
Ah. The was different earlier. With this PR, the condition is no longer required, since the `processBlockedNode` function itself perform required conditional branch. I'll remove this.